verl/workers/megatron_workers.py

```diff
@@ -639,15 +644,31 @@ def __init__(self, config):
             self.config.micro_batch_size //= mpu.get_data_parallel_world_size()
             self.config.micro_batch_size_per_gpu = self.config.micro_batch_size
 
-    def _build_rm_model(self, model_path, megatron_config: ModelParallelConfig, override_model_config):
+    def _build_rm_model(self,
+                        model_path,
+                        megatron_config: ModelParallelConfig,
+                        override_model_config,
+                        sft_tokenizer=None,
+                        rm_tokenizer=None):
         from megatron.core.models.gpt.gpt_model import ModelType
         from verl.utils.model import print_model_size, update_model_config
         from verl.utils.megatron_utils import get_model
         from transformers import AutoModelForCausalLM, AutoTokenizer, AutoConfig
 
         # Step 1: initialize the tokenizer
         local_path = copy_local_path_from_hdfs(model_path)
-        self.tokenizer = hf_tokenizer(local_path)
+        try:
+            self.tokenizer = hf_tokenizer(local_path)
+        except OSError:
+            # If the model path doesn't contain a tokenizer, we use rm_tokenizer or sft_tokenizer
+            if rm_tokenizer:
+                self.tokenizer = rm_tokenizer
+            else:
+                self.tokenizer = sft_tokenizer
```

Comment:
thanks for the contribution! I'd suggest raise the exception on oserror if rm_tokenizer is None, instead of fallback to the sft tokenizer

Reply:
Thank you for your suggestion! I've done it accordingly and raised a `ValueError` instead to make it more accurate. When no tokenizer is found in the model path, `rm_tokenizer` should be provided by the user, and it's not a system error.